src/components/UserForm/UserForm.js

```diff
@@ -0,0 +1,261 @@
+                                    'Invalid WhatsApp number'
+                                )
+                                break
+                            default: {
+                                const field =
+                                    error.errorProperty || FinalForm.FORM_ERROR
+                                errors[field] = error.message
+                            }
+                        }
+                        return errors
+                    },
+                    {}
+                ) || { [FinalForm.FORM_ERROR]: error }
+            )
+        }
+    }
+
+    return (
+        <Form
+            loading={loading}
+            error={error}
+            submitButtonLabel={submitButtonLabel}
+            onSubmit={handleSubmit}
+        >
+            {({ values, submitError }) => (
+                <>
+                    {submitError && (
+                        <NoticeBox
+                            error
+                            title={
+                                user
+                                    ? i18n.t('Error updating user')
+                                    : i18n.t('Error creating user')
+                            }
+                            className={styles.noticeBox}
+                        >
+                            {submitError.message}
+                        </NoticeBox>
+                    )}
+                    <InviteUserSection
+                        user={user}
+                        emailConfigured={emailConfigured}
+                    />
+                    <BasicInformationSection
+                        user={user}
+                        inviteUser={values.inviteUser}
+                        userInterfaceLanguage={userInterfaceLanguage}
+                        interfaceLanguageOptions={interfaceLanguageOptions}
+                        userDatabaseLanguage={userDatabaseLanguage}
+                        databaseLanguageOptions={databaseLanguageOptions}
+                        currentUserId={currentUser.id}
+                    />
+                    <SecuritySection
+                        user={user}
+                        inviteUser={values.inviteUser}
+                        externalAuth={values.externalAuth}
+                        changePassword={values.changePassword}
+                        password={values.password}
+                    />
+                    <ContactDetailsSection user={user} />
+                    <OrganisationUnitsSection user={user} />
+                    <RolesSection
+                        user={user}
+                        userGroupOptions={userGroupOptions}
+                        userRoleOptions={userRoleOptions}
+                    />
+                    <AnalyticsDimensionsRestrictionsSection
+                        user={user}
+                        dimensionConstraintOptions={dimensionConstraintOptions}
+                    />
```

Comment:
The various sections of the user form were broken up into `React.memo`ed components for performance reasons.

In order to support conditional rendering based on the current state of `values`, react final form will rerender the entire form on input/change. While this is fine for smaller forms, this results in very noticeable input lag on larger forms such as this one.